sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,72 @@
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::{Error, BoxDynError};
+use crate::postgres::{PgArgumentBuffer, PgValueFormat, PgTypeInfo, PgValueRef, Postgres};
+use crate::types::Type;
+
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>
+}
+
+impl PgLTree {
+    pub fn push(&mut self, label: String) -> Result<(), Error> {
+        if label.chars().all(|c| c.is_ascii_alphabetic() || c.is_ascii_digit() || c == '_') {
+            self.labels.push(label);
+            Ok(())
+        } else {
+            Err(Error::InvalidLtreeLabel)
+        }
+    }
+}
+
+impl FromStr for PgLTree {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        Ok(
+            Self {
+                labels: s.split('.').map(|s| s.to_owned()).collect()
+            }
+        )
+    }
+}
+
+impl Display for PgLTree {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.labels.join("."))
+    }
+}
+
+impl Type<Postgres> for PgLTree {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::LTREE
+    }
+}
+
+impl Encode<'_, Postgres> for PgLTree {
+    fn encode_by_ref(&self, buf: &mut PgArgumentBuffer) -> IsNull {
+        buf.extend(1i8.to_le_bytes());
+        buf.extend(self.to_string().as_bytes());
+
+        IsNull::No
+    }
+}
+
+impl<'r> Decode<'r, Postgres> for PgLTree {
+    fn decode(value: PgValueRef<'r>) -> Result<Self, BoxDynError> {
+        match value.format() {
+            PgValueFormat::Binary => {
+                let bytes = value.as_bytes()?;
+                let version = i8::from_le_bytes([bytes[0]; 1]);
+                if version != 1 {
+                    todo!("add error here")
+                }
+                Ok(Self::from_str(&String::from_utf8(bytes[1..].to_vec())?)?)
```

Comment:
Could we avoid this allocation by replacing `String` with `std::str` and removing the `.to_vec()`